courses/intro-to-ada/intro_to_ada.rst

```diff
@@ -198,253 +207,427 @@ There are several note worthy things in the above program:
 
-    procedure Greet is
-       I : Integer := 0;
+    procedure Check_Direction is
+       N : Integer;
     begin
        loop
-          --  Expression must be of a discrete type. All the
-          --  values must be covered.
-          case I is
-             when 0 =>
-                Put_Line ("Starting...");
-                Put_Line ("No really");
-                --  You can put several statements in a branch.
-                --  There is no break.
+          Put ("Enter an integer value: ");  -- Puts a String
+          Get (N);  -- Reads an Integer
+          Put (N);  --Puts an Integer
+          case N is
+             when 0 | 360 =>
+                Put_Line (" is due east");
+             when 1..89 =>
+                Put_Line (" is in the northeast quadrant");
+             when 90 =>
+                Put_Line (" is due north");
+             when 91..179 =>
+                Put_Line (" is in the northwest quadrant");
+             when 180 =>
+                Put_Line (" is due west");
+             when 181..269 =>
+                Put_Line (" is in the southwest quadrant");
+             when 270 =>
+                Put_Line (" is due south");
+             when 271..359 =>
+                Put_Line (" is in the southeast quadrant");
+             when others =>
+                Put_Line (" Au revoir");
+                exit;
+          end case;
+       end loop;
+    end Check_Direction;
 
-             when 3 .. 5 =>
-                Put_Line ("Hello");
+This program repeatedly prompts for an Integer value and then, if the value is
+in the range 0..360, displays the associated quadrant or axis.  If the
+value is an Integer outside this range, the loop (and the program) terminate
+after outputting a farewell message.
 
-             when 7 | 9 =>
-                Put_Line ("World");
+The effect of the case statement is similar to the if statement in an earlier
+example, but the case statement can be more efficient because it does not involve
+multiple range tests.
 
-             when 10 =>
-                exit;  -- This exits out of the loop ! Not equivalent to break !
+Notable points about Ada's case statement:
 
-             when others => Put_Line ("I in (1, 2, 6, 8)");
-             --  ‘when others’ must be the last one and alone (if
-             --  present)
-          end case;
-          I := I + 1;
-       end loop;
-    end Greet;
+-  The case expression (here the variable N) must be of a discrete type, i.e.
+   either an integer type or an enumeration type.  Discrete types will
+   be covered in more detail later
+   `discrete types <TODO:linktodiscretetypes>`__.
 
-Notable points about Ada's case statement:
+-  Every possible value for the case expression needs to be covered by a unique
+   branch of the case statement. This will be checked at compile time.
+
+-  A branch can specify a single value, such as ``0``; a range of values,
+   such as ``1..89``; or any combination of the two (separated by a "|"
```

Comment:
Shouldn't we use whitespaces for ranges?

Reply:
Yes!